emperor/support_files/js/draw.js

```diff
@@ -275,32 +279,111 @@ define(['underscore', 'three', 'jquery'], function(_, THREE, $) {
     return arrow;
   }
 
-  function drawTrajectoryLine(trajectory, currentFrame, color, radius) {
+  /**
+   * Returns a new trajectory line dynamic mesh
+   */
+  function drawTrajectoryLineDynamic(trajectory, currentFrame, color, radius) {
     // based on the example described in:
     // https://github.com/mrdoob/three.js/wiki/Drawing-lines
     var material, points = [], lineGeometry, limit = 0, path;
 
-    _trajectory = trajectory.representativeCoordinatesAtIndex(currentFrame);
+    _trajectory = trajectory.representativeInterpolatedCoordinatesAtIndex(
+                                                                  currentFrame);
+    if (_trajectory === null || _trajectory.length == 0)
+      return null;
 
-    material = new THREE.MeshPhongMaterial({color: color});
-    material.matrixAutoUpdate = true;
-    material.transparent = false;
+    material = new THREE.MeshPhongMaterial({
+        color: color,
+        transparent: false});
 
     for (var index = 0; index < _trajectory.length; index++) {
       points.push(new THREE.Vector3(_trajectory[index].x,
                   _trajectory[index].y, _trajectory[index].z));
     }
 
     path = new THREE.EmperorTrajectory(points);
+
     // the line will contain the two vertices and the described material
     // we increase the number of points to have a smoother transition on
     // edges i. e. where the trajectory changes the direction it is going
-    lineGeometry = new THREE.TubeGeometry(path, (points.length - 1) * 3, radius,
-                                          10, false);
+    lineGeometry = new THREE.TubeGeometry(path,
+                                    (points.length - 1) * NUM_TUBE_SEGMENTS,
+                                    radius,
+                                    NUM_TUBE_CROSS_SECTION_POINTS,
+                                    false);
 
     return new THREE.Mesh(lineGeometry, material);
   }
 
+  /**
+   * Disposes a trajectory line dynamic mesh
+   */
+  function disposeTrajectoryLineDynamic(mesh) {
```

Comment:
Both of the dispose functions do exactly the same thing, perhaps worth merging them into `disposeTrajectoryLine`?

Reply:
Yeah, I was torn on this too.  I split them because the underlying geometry and material objects inside of the mesh are what needs to be disposed, and those are both shareable across mesh instances.  I was leaning towards a pattern where we put dispose (destructor) next to draw (constructor), so that changes in one function can be easily mirrored to the other.  If we have no short term plan to add geometry or material reuse and want to add as a rule instead that these are copied in every mesh, I think we should use a single function for disposing arbitrary meshes, which I think is just as good.  

On a side note, we're technically leaking Axis meshes when switching axes, so I figured a dispose method per type of scene object made sense when I realized I didn't want to trace the code to find out how axis meshes were initialized.